### ISSUE TITLE
Recreate the Europe Map

### ISSUE THREAD

**Issue by pgrawehr** [aAXEe/online_chart_ol3#233]
Would it be possible to recreate the Europe map? Currently, both the KAP format and the mbtiles format only contain level 16 for this map, and no lower levels. This causes strange issues both in OpenCPN as well as in Signalk/Freeboard-sk, where nothing is shown until exactly that level is reached. Other maps (like the one for the baltic sea) does not have this problem, although it could start at lower levels as well. 

To be really usable, the maps should have all levels between 2 and at least 16 (better 17 or 18 for harbours), or maybe every second level in-between. 

I could create the maps myself, but I want to avoid downloading that many tiles just for myself. (I'll eventually need all european waters from St. Petersburg down to Athens)

**Comment by stevo01** [comment]
Hi
> Would it be possible to recreate the Europe map? Currently, both the KAP format and the mbtiles format only contain level 16 for this map, and no lower levels. This causes strange issues both in OpenCPN as well as in Signalk/Freeboard-sk, where nothing is shown until exactly that level is reached.

Thanks for your feedback. The maps are based on MOBAC project files stored here: https://github.com/stevo01/SeaChartCreator/tree/master/sample/atlas/mobac/inland-water-ways
Feel free to update the files and send me pull request. After a review I can integrate your changes and build new kap and mbtile files.

>Other maps (like the one for the baltic sea) does not have this problem, although it could start at lower levels as well.

The reason is that the project files come from different authors. The authors used different tools and strategies to design the maps. Biggest challenge is to reduce the number of needed tiles and cover everything whats important for the navigator. 

> To be really usable, the maps should have all levels between 2 and at least 16 (better 17 or 18 for harbours), or maybe every second level in-between.

There is a wiki article available which describes how to "design a chart":
https://wiki.openseamap.org/wiki/OpenSeaMap-dev:De:Chart-Designer

**Comment by pgrawehr** [comment]
Thanks for the links, I'll have a look at that.

**Comment by pgrawehr** [comment]
Question: Can your tool handle non-square (polygon-shaped) areas, as MOBAC can?
Just curious: What's the advantage of your tool over directly using Mobac?

**Comment by stevo01** [comment]
> Question: Can your tool handle non-square (polygon-shaped) areas, as MOBAC can?
> Just curious: What's the advantage of your tool over directly using Mobac?

Honestly, I haven't figured that out yet. The "Chart Designer" is based on Mobac and most functions are available in both applications.

**Comment by pgrawehr** [comment]
@stevo01 I've done [Lake Constance](https://github.com/pgrawehr/SeaChartCreator/tree/EuropeUpdate) as a first start. It works fine with SignalK, but OpenCPN only displays one single small section of the map with only one resolution. Any idea how to analyze that?

(I would upload my created map, but Github doesn't like anything > 10MB 👎 )

**Comment by stevo01** [comment]
@pgrawehr: have you generated/tested mbtiles or kap files? I will generate both and give you feedback...

**Comment by pgrawehr** [comment]
I have created mbtiles, because that works in both applications (or is supposed to work in both...)

**Comment by stevo01** [comment]
@pgrawehr: next question: Which application do you used for generation of the mbtiles?

**Comment by stevo01** [comment]
I generated mbtiles with OpenSeaMap today. The size of the File is around 50MB. The parts marked as polygon are not included because the SeaChartCreator Application processes only the "map" elements and not the "PolygonMap" from mobac project files. 

OpenSeaMap in version 5.0.0 is able to open the map but shows just a small part of the map in lower zoom level. (I think there is a bug in my mbtile generator) The presentation of the map in high zoom level is bad because the areas defined as "polygon" are missing".

![konstanz](https://user-images.githubusercontent.com/33812539/72274766-08f62a00-362d-11ea-9538-847c1bef43da.png)

Can you replace the "polygon maps"? I did not support polygons because I cannot use them for generation of KAP files...

**Comment by pgrawehr** [comment]
I used Mobac for generating the mbtiles file. OpenCPN (From the screenshot that's what you're using) just shows one very small tile of the whole map for me. I think it has nothing to do with the polygon regions, but I'll try to eliminate those.

**Comment by stevo01** [comment]
> I used Mobac for generating the mbtiles file. OpenCPN (From the screenshot that's what you're using) just shows one very small tile of the whole map for me. I think it has nothing to do with the polygon regions, but I'll try to eliminate those.

👍

**Comment by pgrawehr** [comment]
@stevo01 I've added an XML that uses only rectangles now. Needs a bit more tiles, but should do the trick. Again, I've used mobac to create the Mbtiles file. Still the issue for me remains: In OpenCPN, only a small fraction of the map is visible, the SignalK chart server is rendering the maps correctly. I do not know how to create KAP with Mobac, it doesn't seem to be in the list of available formats.

**Comment by stevo01** [comment]
I generated a mbtile file with scripts from SeaChartCreator project and made upload to ftp server.
The file is available for download:
https://ftp5.gwdg.de/pub/misc/openstreetmap/openseamap/charts/mbtiles/test/OSM-OpenCPN2-BodenseeTest.mbtiles

note: OpenCPN seems to work with the file.

**Comment by pgrawehr** [comment]
@stevo01 : Yes, this map works with both tools I have tried (OpenCPN and SignalK). Not sure what is different to the one I generated myself. I'll investigate later, when I find time. 

I'll go ahead then and prepare the XML for other regions that I think need updating/fixing.

**Comment by stevo01** [comment]
Ok. Thanks for support. I plan to review the map with other OSM team member and give you feedback next days.

**Comment by stevo01** [comment]
@pgrawehr: please send me your telephone number and / or email to project@openseamap.org.

**Comment by stevo01** [comment]
I created and tested new map for Lake Constance. This map includes maps in different zoom levels and each zoom level covers complete area of the lake. Download: https://ftp5.gwdg.de/pub/misc/openstreetmap/openseamap/charts/kap/OSM-OpenCPN2-KAP-LakeConstance.7z
